docs/plugin-protocol/tfplugin5.proto

```diff
@@ -811,3 +819,45 @@ message CallFunction {
         FunctionError error = 2;
     }
 }
+
+message ListResource {
+    message Request {
+        // type_name is the list resource type name.
+        string type_name = 1;
+
+        // configuration is the list ConfigSchema-based configuration data.
+        DynamicValue config = 2;
+    }
+
+    message Event {
+        // Resource contains all data associated with each resource instance.
+        message Result {
+            // identity is the resource identity data of the resource instance.
+            ResourceIdentityData identity = 1;
+
+            // display_name can be displayed in a UI to make it easier for humans to identify a resource
+            string display_name = 2;
+
+            // optional resource object which can be useful when combining list blocks in configuration
+            DynamicValue resource_object = 3;
```

Comment:
Checking my assumptions:

Is it completely the provider's decision re: whether or not to return `resource_object`? Possibly via information in the list config, as defined in the list config schema?

Or do we want the request message to have an explicit `include` (or similar) field that is not schema-dependent?

Reply:
The current idea is to have an attribute in the config (maybe something like `include_resource`) and, if that field is set, the provider should include the `resource_object`.

Should this be made more explicit with an additional field in the proto request (outside the config)? I imagine that having a field at protocol level removes any ambiguity

Reply:
yes, if resource data is optional, we should probably directly include whether it's requested or not in the request.

Reply:
If we want that attribute standardized, then perhaps it should be in the proto definition

Reply:
If it is in the proto file, what does it mean to the sdk/framework when the attribute is set? Would the sdk invalidate a provider response that does not include the resource despite it being requested for, or is the attribute simply a suggestion?

Reply:
> If it is in the proto file, what does it mean to the sdk/framework when the attribute is set? Would the sdk invalidate a provider response that does not include the resource despite it being requested for, or is the attribute simply a suggestion?

I imagine the SDK/Framework would simply pass this along to the provider implementation, without inspecting the response.

We _could_ raise a "provider implementation error" if the attribute is set & the returned resource is absent. I don't see a lot of value in that; open to changing my opinion on it, though.

Reply:
I will add a `include_resource_object` fields to the request.